cli/flox-rust-sdk/src/models/search.rs

```diff
@@ -156,14 +156,22 @@ pub struct Query {
     pub match_name_or_rel_path: Option<String>,
     /// Limit search results to a specified number
     pub limit: Option<u8>,
+    /// Return a single result for each package descriptor used by `search` and
+    /// `install`.
+    pub deduplicate: bool,
 }
 
 impl Query {
     /// Construct a query from a search term and an optional search result limit.
-    pub fn from_term_and_limit(
+    ///
+    /// `deduplicate = true` will return unique results for flox search;
+    /// for example, a single result will be returned for a package even if
+    /// there are multiple versions or systems of a package.
```

Comment:
suggestion(nb): mention what information is lost when merging packages together

```suggestion
    ///
    /// `deduplicate = true` will return unique results for flox search;
    /// for example, a single result will be returned for a package even if
    /// there are multiple versions or systems of a package.
    /// This is assuming, that two packages with the same path are 
    /// 1) indeed the same package
    /// 2) meta field values do not depend on the system
```

Reply:
Tried to justify the assumption a bit:
```
    /// This assumes that the only data needed from a package is it's relPath,
    /// and it assumes all packages with the same relPath should be treated as
    /// one.
    /// Although hypothetically x86_64-linux.hello and aarch64-darwin.hello
    /// could be "different" packages, since we only want to print hello once in
    /// search results, this is a valid assumption.
```

Reply:
Also amending something about `description`